llvm/test/tools/sycl-post-link/device-requirements/reqd-sub-group-size.ll

```diff
@@ -0,0 +1,194 @@
+; Original code:
+; #include <sycl/sycl.hpp>
+
+; int main() {
+;   sycl::queue q;
+;   q.submit([&](sycl::handler &h) {
+;       h.parallel_for<class KernelA>(
+;           sycl::range<1>(32),
+;           [=](sycl::item<1> it) [[sycl::reqd_sub_group_size(16)]] {});
+;     });
+;   q.submit([&](sycl::handler &h) {
+;       h.parallel_for<class KernelB>(
+;           sycl::range<1>(32),
+;           [=](sycl::item<1> it) [[sycl::reqd_sub_group_size(32)]] {});
+;     });
+;   q.submit([&](sycl::handler &h) {
+;       h.parallel_for<class KernelC>(
+;           sycl::range<1>(32),
+;           [=](sycl::item<1> it) [[sycl::reqd_sub_group_size(16)]] {});
+;     });
+;   return 0;
+; }
+
+; RUN: sycl-post-link -split=auto %s -o %t.table
+; RUN: FileCheck %s -input-file=%t_0.prop --check-prefix CHECK-PROP-AUTO-SPLIT-0
+; RUN: FileCheck %s -input-file=%t_1.prop --check-prefix CHECK-PROP-AUTO-SPLIT-1
+
+; CHECK-PROP-AUTO-SPLIT-0: [SYCL/device requirements]
+; CHECK-PROP-AUTO-SPLIT-0: reqd_sub_group_size=2|gAAAAAAAAAAIAAAA
+
+; CHECK-PROP-AUTO-SPLIT-1: [SYCL/device requirements]
+; CHECK-PROP-AUTO-SPLIT-1: reqd_sub_group_size=2|gAAAAAAAAAAEAAAA
+
+; ModuleID = '/tmp/source-706237.bc'
+source_filename = "llvm-link"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+$_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE_clES4_E7KernelAEE = comdat any
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_E7KernelA = comdat any
+
+$_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE0_clES4_E7KernelBEE = comdat any
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE0_clES2_E7KernelB = comdat any
+
+$_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE1_clES4_E7KernelCEE = comdat any
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE1_clES2_E7KernelC = comdat any
+
+@__spirv_BuiltInGlobalInvocationId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+
+; Function Attrs: norecurse
+define weak_odr dso_local spir_kernel void @_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE_clES4_E7KernelAEE() local_unnamed_addr #0 comdat !srcloc !46 !kernel_arg_buffer_location !47 !intel_reqd_sub_group_size !48 !sycl_fixed_targets !49 !sycl_kernel_omit_args !50 {
```

Comment:
To further simply this IR, you can compile your code with `-D__SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING__`, this will remove all those `__pf_kernel_wrapper` kernels and associated aliasing metadata.

I would also suggest to add a command you used to compile the source code into the test, so it can be easily re-generated later